packages/select/src/components/multi-select/multiSelect.tsx

```diff
@@ -406,6 +471,31 @@ export class MultiSelect<T> extends AbstractPureComponent<MultiSelectProps<T>, M
         };
     };
 
+    private getKeyDownHandler = (handleQueryListKeyDown: React.KeyboardEventHandler<HTMLElement>) => {
+        return (e: React.KeyboardEvent<HTMLElement>) => {
+            if (e.key === "Escape" || e.key === "Tab") {
+                if (this.popoverInputForCustomTarget != null) {
+                    this.popoverInputForCustomTarget.blur();
+                }
+                this.setState({ isOpen: false });
```

Comment:
why exclude the bit that opens the popover on other keys? I think that's important for keyboard nav